src/engine/blst.rs

```diff
@@ -0,0 +1,366 @@
+        "4A6FB95F0241B3583771E799120C87AAE3C843ECDB50A38254A92E198968922F",
+        "1005079F96EC412A719FE2E9FA67D421D98FB4DEC4181459E59430F5D502BD2A",
+        "64960B8692062DCB01C0FFBAC569478A89AD880ED3C9DF710BED5CE75F484693",
+        "03C2882155A447642BD21FB1CF2553F80955713F09BBBBD9724E2CBFD8B19D41",
+        "0AB07FECB59EE3435F6129FCD602CB519E56D7B426941633E37A3B676A24830F",
+        "12FA5861459EFFBAE654827D98BFDFEA5545DDF8BB9628579463DA21F17462B5",
+        "6A6296A0376D807530DB09DC8BB069FFDEC3D7541497B82C722A199D6B7C5B06",
+        "153D2C81B54D7E1C3E83EA61C7F66FD88155F1713EE581E2BE8438CA9FEE1A02",
+        "216BCCC4AE97FE3E1D4B21C375C46140FA153E7868201A43480889047ACD0C2D",
+        "381BD4FE924EB10E08F2A227D3DB2083AA0E5A1F661CD3C702C4B8A9385E7839",
+        "723A7640FD7E65473131563AB5514916AC861C2695CE6513E5061E597E5E1A81",
+    ];
+
+    const TEST_POINT: &str = "456006fff56412d329d527901d02877a581a89cfa677ca963eb9d680766234cc";
+    const TEST_EVAL: &str = "29732a1e0e074ab05ee6a9e57794c5ad1965b98b6c8c6ecde96ac776ea06ff5b";
+
+    const EXPECTED_COMMITMENT: &str =  "8424fb9dc224ab79efccf6710edea3b936d03bbd323f052bb9c4b2efe9f98239e7c3e48148f243065cee910054a10e71";
+    const EXPECTED_PROOF: &str = "895cdfe1bf26bbf10bdc0d90178ec89635269cca7c9b39836a76e91689ad3fa4d1772f8d60cdd86cd4bfd1dedbdec81d";
+
+    impl BlstBackend {
+        pub fn random_poly(&self) -> FsPoly {
+            let mut poly = FsPoly::new(self.fft_settings.get_max_width());
+            for i in 0..self.fft_settings.get_max_width() {
+                poly.set_coeff_at(i, &Fr::rand());
+            }
+            poly
+        }
+
+        pub fn random_fr(&self) -> FsFr {
+            FsFr::rand()
+        }
+
+        pub fn random_g1(&self) -> FsG1 {
+            FsG1::rand()
+        }
+    }
+
+    #[test]
+    #[tracing_test::traced_test]
+    fn test_arkworks_backend() {
+        let cfg = crate::engine::backend::BackendConfig::new(Some(4), None);
+        let backend = BlstBackend::new(Some(cfg.clone()));
+        let poly = backend.random_poly();
+        tracing::info!("poly: {:?}", poly.clone());
+        let x = backend.random_fr();
+        let y = poly.eval(&x);
+        let commitment = backend.commit_to_poly(poly.clone()).unwrap();
+        let proof = backend.compute_proof_single(poly.clone(), x).unwrap();
+        let result = backend
+            .verify_proof_single(proof, x, y, commitment)
+            .unwrap();
+        assert!(result);
+    }
+
+    #[test]
+    fn test_arkworks_g1_serialize_deserialize() {
+        let cfg = crate::engine::backend::BackendConfig::new(Some(4), None);
+        let backend = BlstBackend::new(Some(cfg.clone()));
+        let g1 = backend.random_g1();
+        println!("g1: {:?}", g1);
+        let g1_bytes = g1.to_bytes();
+        let reserialized = FsG1::from_bytes(&g1_bytes);
+        assert_eq!(reserialized, Ok(g1));
+    }
+
+    // TODO: make proper test for fixed setup file and fixed secret
```

Comment:
I think this can also just call the `write_setup` function, and then delete the file at the end of the test. maybe a smaller degree than 2^20 though